src/System.Runtime/tests/System/Lazy.cs

```diff
@@ -3,69 +3,113 @@
+            Assert.Throws<ArgumentNullException>("valueFactory", () => new Lazy<object>(null)); // Value factory is null
+        }
+
+        [Fact]
+        public static void TestCtor_LazyThreadSafetyMode()
+        {
+            var lazyString = new Lazy<string>(LazyThreadSafetyMode.PublicationOnly);
+            VerifyLazy(lazyString, "", false);
+        }
+
+        [Fact]
+        public static void TestCtor_LazyThreadSafetyMode_Invalid()
+        {
+            Assert.Throws<ArgumentOutOfRangeException>("mode", () => new Lazy<string>(LazyThreadSafetyMode.None - 1)); // Invalid thread saftety mode
+            Assert.Throws<ArgumentOutOfRangeException>("mode", () => new Lazy<string>(LazyThreadSafetyMode.ExecutionAndPublication + 1)); // Invalid thread saftety mode
+        }
+
+        [Fact]
+        public static void TestCtor_ValueFactor_Bool()
+        {
+            var lazyString = new Lazy<string>(() => "foo", false);
+            VerifyLazy(lazyString, "foo", true);
+
+            var lazyInt = new Lazy<int>(() => 1, false);
+            VerifyLazy(lazyInt, 1, true);
+        }
+
+        [Fact]
+        public static void TestCtor_ValueFactory_Bool_Invalid()
+        {
+            Assert.Throws<ArgumentNullException>("valueFactory", () => new Lazy<object>(null, false)); // Value factory is null
+        }
+
+        [Fact]
+        public static void TestCtor_ValueFactor_LazyThreadSafetyMode()
+        {
+            var lazyString = new Lazy<string>(() => "foo", LazyThreadSafetyMode.PublicationOnly);
+            VerifyLazy(lazyString, "foo", true);
+
+            var lazyInt = new Lazy<int>(() => 1, LazyThreadSafetyMode.PublicationOnly);
+            VerifyLazy(lazyInt, 1, true);
+        }
+
+        [Fact]
+        public static void TestCtor_ValueFactor_LazyThreadSafetyMode_Invalid()
+        {
+            Assert.Throws<ArgumentNullException>("valueFactory", () => new Lazy<object>(null, LazyThreadSafetyMode.PublicationOnly)); // Value factory is null
+
+            Assert.Throws<ArgumentOutOfRangeException>("mode", () => new Lazy<string>(() => "foo", LazyThreadSafetyMode.None - 1)); // Invalid thread saftety mode
+            Assert.Throws<ArgumentOutOfRangeException>("mode", () => new Lazy<string>(() => "foof", LazyThreadSafetyMode.ExecutionAndPublication + 1)); // Invalid thread saftety mode
+        }
+
+        [Fact]
+        public static void TestToString_DoesntForceAllocation()
+        {
+            var lazy = new Lazy<object>(() => 1);
+            Assert.NotEqual("1", lazy.ToString());
+            Assert.False(lazy.IsValueCreated);
+
+            object tmp = lazy.Value;
+            Assert.Equal("1", lazy.ToString());
         }
 
         [Fact]
-        public static void TestValueExceptions()
+        public static void TestValue_Invalid()
         {
             string value = "Test";
-            Lazy<string> lazy = new Lazy<string>(() => value);
+            var lazy = new Lazy<string>(() => value);
```

Comment:
We need this instance to avoid a compiler error - we get an error saying something like "use of unassigned variable lazy"
Although we know that the delegate action will be called after the ctor is done (when lazy.Value is called), the C# compiler doesn't know so this is a workaround